drivers/i2s/i2s_ll_stm32.c

```diff
@@ -349,9 +359,30 @@ static int i2s_stm32_trigger(const struct device *dev, enum i2s_dir dir,
 			LOG_ERR("DRAIN trigger: invalid state");
 			return -EIO;
 		}
-		stream->stream_disable(stream, dev);
-		stream->queue_drop(stream);
-		stream->state = I2S_STATE_READY;
+
+		if (dir == I2S_DIR_TX) {
+			if (stream->state == I2S_STATE_RUNNING) {
+				stream->state = I2S_STATE_STOPPING;
+				/*
+				 * Indicate that the transition to I2S_STATE_STOPPING
+				 * is triggered by DRAIN command
+				 */
+				stream->sample_tx_stop_drain = true;
+			} else if ((queue_is_empty(&stream->mem_block_queue) == false) ||
+						(LL_SPI_IsActiveFlag_TXC(cfg->i2s))) {
```

Comment:
An abstraction function will be required here to deal with F4/H5 variations:
```
static inline uint32_t ll_func_i2s_is_busy(I2S_TypeDef *i2s)
{
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_i2s)
	return LL_SPI_IsActiveFlag_TXC(i2s) == 0;
#else
	return LL_SPI_IsActiveFlag_BSY(i2s);
#endif /* st_stm32h7_i2s */
```

Reply:
yes i forget to commit this modification :( i will do it now